core/trino-main/src/main/java/io/trino/dispatcher/LocalDispatchQueryFactory.java

```diff
@@ -93,6 +95,7 @@ public LocalDispatchQueryFactory(
         this.warningCollectorFactory = requireNonNull(warningCollectorFactory, "warningCollectorFactory is null");
         this.clusterSizeMonitor = requireNonNull(clusterSizeMonitor, "clusterSizeMonitor is null");
         this.executor = dispatchExecutor.getExecutor();
+        this.maxStateMachineThreadsPerQuery = dispatchExecutor.getMaxStateMachineThreadsPerQuery();
```

Comment:
Maybe simply inject the `QueryManagerConfig` here?

Reply:
Yeah, that is cleaner. Changed.